.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,43 @@
+name: Static Analysis
+
+on:
+  pull_request:
+    branches:
+      - master
+    paths:
+      - .github/workflows/static-analysis.yml
+      - composer.*
+      - lib/**
+      - phpstan*
+
+  push:
+    branches:
+      - master
+    paths:
+      - .github/workflows/static-analysis.yml
+      - composer.*
+      - lib/**
+      - phpstan*
+
+jobs:
+  static-analysis-phpstan:
+    name: Static Analysis with PHPStan
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: none
+          php-version: 8.3
+          tools: cs2pr
+
+      - name: Install dependencies with Composer
```

Comment:
I would suggest to simplify this action and introduce a composer cache like suggested here:

https://github.com/shivammathur/setup-php?tab=readme-ov-file#cache-composer-dependencies

Something like:
```
- name: Get composer cache directory
  id: composer-cache
  run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT

- name: Cache dependencies
  uses: actions/cache@v4
  with:
    path: ${{ steps.composer-cache.outputs.dir }}
    key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
    restore-keys: ${{ runner.os }}-composer-

- name: Install dependencies
  run: composer install --prefer-dist```